source/features/remove-is-open-is-closed.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import features from '../libs/features';
+
+const countMatches = (string: string, regex: RegExp): number => {
+	return ((string || '').match(regex) ?? []).length;
+};
+
+function createMergeLink(): HTMLAnchorElement {
+	const linkMergedSearchParameters = new URLSearchParams(location.search);
+	const linkIsMerged: HTMLAnchorElement = (
+		<a href="" className="btn-link">
+		Merged
+		</a>
+	);
+	const regexpQueryTotal = /is:open|is:closed|is:issue/g;
+	const regexpQuery = /is:open|is:closed|is:issue/;
+
+	let linkMergedSearchString = new URLSearchParams(location.search).get('q') ?? select<HTMLInputElement>('#js-issues-search')!.value;
+
+	while (countMatches(linkMergedSearchString, regexpQueryTotal) > 1) {
+		linkMergedSearchString = linkMergedSearchString.replace(regexpQuery, '').trim();
+	}
+
+	if (countMatches(linkMergedSearchString, /is:merged/) === 1) {
+		linkMergedSearchParameters.set('q', linkMergedSearchString.replace(/is:merged/, 'is:issue').trim());
+		linkIsMerged.classList.add('selected');
+	} else if (countMatches(linkMergedSearchString, regexpQueryTotal) === 1) {
+		linkMergedSearchParameters.set('q', linkMergedSearchString.replace(regexpQuery, 'is:merged').trim());
+	}
+
+	linkIsMerged.search = String(linkMergedSearchParameters);
+
+	return linkIsMerged;
+}
+
+function init(): void {
+	const divTableListFiltersParent = select('div.table-list-filters');
+	const inputJsIssuesSearch = select<HTMLInputElement>('#js-issues-search');
+	if ((divTableListFiltersParent === null) || (inputJsIssuesSearch === null)) {
+		return;
+	}
+
+	const mergeLink = createMergeLink();
+
+	const containerTargetButtons = divTableListFiltersParent.children[0].children[0];
+	const targetButtons: HTMLCollectionOf<HTMLAnchorElement> = containerTargetButtons.children;
```

Comment:
Use selectors for these two variables

Reply:
I don't have certainty that on the page there would be another link (anchor) with class .btn-link. That's why I wanted to grab first the container and then manually get to the anchor elements.

After reading your second comment (the one I put the eyes on) I have the idea that I can grab the divTableListFiltersParent, and then using `select('.btn-link', divTableListFiltersParent)` I can get to the anchor elements I need to take.

What do you think? It's the right approach to get the elements? Or it's better to directly grab them without touching the container element?

Reply:
Generally speaking, if you have a container ready, yes you can just look into it. 

Alternatively, maybe you can skip creating a variable for that container and just use the full selector where needed. 

In this case (I didn’t look at the actual DOM) you can achieve the same result with this selector `.table-list-filters > :first-child > :first-child` but surely there are better classes to target.

Reply:
I didn't make any changes, I understand that it is better to not create a new variable. However, I need the container to iterate over it's elements and change it's behaviours (open & close links), and secondly to add the new mergeLink anchor element.

I would be glad to hear more feedback if you could see the code, and tell me if it is better to create the variable with container, or if it's better to search twice the dom using `.table-list-filters > :first-child > :first-child` under the actual circumstances.

Best regards

Reply:
Rather than discussing about variable reuse etc, think more in terms of:
- does this _need_ to exist? Is the code more readable as a consequence of it existing?
- is this code split in reasonable parts? Don't add a piece here and another in a different function, if they're working for the same purpose. Don't look for an element here and then pass it to another function which will just work **around** it.

As you designed them, both `addMergeLink` and `togglableFilters` accept a container _inside which_ you'll be doing work. That's the opposite of what functions normally do. If you pass an element to a function, generally I expect it to work on the element specifically, _not_ to be a "close enough" reference to the work you're about to do.

Refer to 872cd8a7 and 81e41219